app/models.py

```diff
@@ -1394,10 +1394,12 @@ def __init__(self,
         try:
             file_exists = Files.query.filter_by(request_id=request_id, hash=hash_).one_or_none()
             if file_exists is not None:
```

Comment:
Instead of doing another query, you can just use `file_exists.deleted`.

`if file_exists is not None and not file_exists.deleted:`